api/service/stagedsync/sync_config.go

```diff
@@ -0,0 +1,378 @@
+package stagedsync
+
+import (
+	"bytes"
+	"encoding/hex"
+	"errors"
+	"math/rand"
+	"reflect"
+	"sort"
+	"sync"
+
+	"github.com/harmony-one/harmony/api/service/legacysync/downloader"
+	pb "github.com/harmony-one/harmony/api/service/legacysync/downloader/proto"
+	"github.com/harmony-one/harmony/core/types"
+	"github.com/harmony-one/harmony/internal/utils"
+)
+
+// Constants for syncing.
+const (
+	downloadBlocksRetryLimit        = 10 // downloadBlocks service retry limit
+	RegistrationNumber              = 3
+	SyncingPortDifference           = 3000
+	inSyncThreshold                 = 0   // when peerBlockHeight - myBlockHeight <= inSyncThreshold, it's ready to join consensus
+	SyncLoopBatchSize        uint32 = 30  // maximum size for one query of block hashes
```

Comment:
Is this too small?

Reply:
I think so. But the legacy downloader is using the hard-coded batch size (which is 30) to not allow other peers to overload DNS nodes. So, the sender checks the requested batch size. So, even if we increase this number, we have to update all DNS nodes as well.